apps/common/lib/lexical/ast/analysis.ex

```diff
@@ -365,6 +367,40 @@ defmodule Lexical.Ast.Analysis do
     State.push_import(state, Import.new(state.document, quoted, module))
   end
 
+  # require of a module using as
+  defp analyze_node(
+         {:require, _meta, [{:__aliases__, _, module}, options]} = quoted,
+         state
+       ) do
+    {_, as_module} =
+      Macro.prewalk(options, nil, fn
+        {:__aliases__, _, as} = ast, nil ->
+          {ast, as}
+
+        ast, acc ->
+          {ast, acc}
+      end)
+
+    State.push_require(state, Require.new(state.document, quoted, module, as_module))
```

Comment:
We have a `fetch_alias_as` helper defined in this module.

```suggestion
    case fetch_alias_as(options) do
      {:ok, as_module} ->
        State.push_require(state, Require.new(state.document, quoted, module, as_module)
        
      :error ->
        state
    end
```